File: nutrio_app/lib/models/pill_reminder.dart
class PillReminder {
  final int? id;
  final int userId;
  final String name;
  final String? dosage;
  final String? comment;
  final String time; // формат HH:mm
  final List<int> daysOfWeek; // 1=Пн, 7=Вс
  final bool active;

  PillReminder({
    this.id,
    required this.userId,
    required this.name,
    this.dosage,
    this.comment,
    required this.time,
    required this.daysOfWeek,
    this.active = true,
  });

  factory PillReminder.fromJson(Map<String, dynamic> json) => PillReminder(
        id: json['id'],
        userId: json['userId'],
        name: json['name'],
        dosage: json['dosage'],
        comment: json['comment'],
        time: json['time'],
        daysOfWeek: List<int>.from(json['daysOfWeek']),
        active: json['active'] ?? true,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'userId': userId,
        'name': name,
        'dosage': dosage,
        'comment': comment,
        'time': time,
        'daysOfWeek': daysOfWeek,
        'active': active,
      };
} 

File: src/main/java/com/nutrio/model/PillReminder.java
package com.nutrio.model;

import jakarta.persistence.*;
import java.time.LocalTime;
import java.util.List;

@Entity
public class PillReminder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private String name; // Название таблетки
    private String dosage; // Дозировка (опционально)
    private String comment; // Комментарий (опционально)
    private LocalTime time; // Время приёма
    private boolean active = true;

    @ElementCollection(fetch = FetchType.EAGER)
    private List<Integer> daysOfWeek; // 1=Пн, 7=Вс

    public PillReminder() {}

    public PillReminder(Long userId, String name, String dosage, String comment, LocalTime time, List<Integer> daysOfWeek, boolean active) {
        this.userId = userId;
        this.name = name;
        this.dosage = dosage;
        this.comment = comment;
        this.time = time;
        this.daysOfWeek = daysOfWeek;
        this.active = active;
    }

    // Геттеры и сеттеры
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDosage() { return dosage; }
    public void setDosage(String dosage) { this.dosage = dosage; }

    public String getComment() { return comment; }
    public void setComment(String comment) { this.comment = comment; }

    public LocalTime getTime() { return time; }
    public void setTime(LocalTime time) { this.time = time; }

    public List<Integer> getDaysOfWeek() { return daysOfWeek; }
    public void setDaysOfWeek(List<Integer> daysOfWeek) { this.daysOfWeek = daysOfWeek; }

    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
} 